### PR TITLE
identity - support `SystemAssignedUserAssigned`

### PR DESCRIPTION
**Pull request by magodo**
Identity supports `SystemAssignedUserAssigned` variant. Also adopt the linux/windows vm to approve.

## Test

```shell
💢 TF_ACC=1 go test -v -timeout=8h ./azurerm/internal/services/compute -run="TestAccLinuxVirtualMachine_identityUpdate"
2021/06/16 17:29:41 [DEBUG] not using binary driver name, it's no longer needed
2021/06/16 17:29:41 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccLinuxVirtualMachine_identityUpdate
=== PAUSE TestAccLinuxVirtualMachine_identityUpdate
=== CONT  TestAccLinuxVirtualMachine_identityUpdate
--- PASS: TestAccLinuxVirtualMachine_identityUpdate (1044.86s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute     1045.006s
```

### CODE REVIEW

File: azurerm/internal/services/compute/virtual_machine.go [hunk 1]
```diff
@@ -68,110 +68,50 @@ func flattenVirtualMachineAdditionalCapabilities(input *compute.AdditionalCapabi
...
-
-	identity := compute.VirtualMachineIdentity{
-		Type: compute.ResourceIdentityType(raw["type"].(string)),
-	}
-
-	identityIdsRaw := raw["identity_ids"].(*pluginsdk.Set).List()
-	identityIds := make(map[string]*compute.VirtualMachineIdentityUserAssignedIdentitiesValue)
-	for _, v := range identityIdsRaw {
-		identityIds[v.(string)] = &compute.VirtualMachineIdentityUserAssignedIdentitiesValue{}
+	config, err := identity.SystemAssignedUserAssigned{}.Expand(input)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/virtual_machine.go:75]: as mentioned in the README for Identity, can be type alias this above? e.g.

```suggestion
	config, err := identity.SystemAssignedUserAssigned{}.Expand(input)
```
```suggestion
	config, err := virtualMachineIdentity{}.Expand(input)
```

it means these are defined in a single place should this need changing

File: azurerm/internal/services/compute/virtual_machine.go [hunk 2]
```diff
@@ -68,110 +68,50 @@ func flattenVirtualMachineAdditionalCapabilities(input *compute.AdditionalCapabi
...
-
-	return []interface{}{
-		map[string]interface{}{
-			"type":         string(input.Type),
-			"identity_ids": identityIds,
-			"principal_id": principalId,
-			"tenant_id":    tenantId,
-		},
-	}, nil
+	return identity.SystemAssignedUserAssigned{}.Flatten(config), nil
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/virtual_machine.go:114]: (as above)

File: azurerm/internal/services/compute/virtual_machine.go [hunk 3]
```diff
@@ -68,110 +68,50 @@ func flattenVirtualMachineAdditionalCapabilities(input *compute.AdditionalCapabi
...
-				},
-
-				"tenant_id": {
-					Type:     pluginsdk.TypeString,
-					Computed: true,
-				},
-			},
-		},
-	}
+	return identity.SystemAssignedUserAssigned{}.Schema()
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/virtual_machine.go:71]: (as above) which probably means we could remove this method and use that directly?

**Review by tombuildsstuff — APPROVED** [review]
hey @magodo 

Thanks for this PR.

Taking a look through if we can add a Type Alias for this then this should otherwise be good to go 👍 

Thanks!

**Comment by magodo** [comment]
@tombuildsstuff Thank you for the review! I've updated accordingly.

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @magodo

**Pull request merged**